osu.Game/Configuration/SettingSourceAttribute.cs

```diff
@@ -24,18 +29,21 @@ public class SettingSourceAttribute : Attribute
 
         public string Description { get; }
 
-        public SettingSourceAttribute(string label, string description = null)
+        public int OrderPosition { get; }
+
+        public SettingSourceAttribute(string label, string description = null, int orderPosition = -1)
```

Comment:
How about making this nullable instead, and allowing negative ordering?

It's pretty arbitrary that the base mod has orders `1, 2` and the derived mods have orders `0, 3`. Meaning if there was a mod that has multiple settings before the `1`, it couldn't do it due to the `> -1` limitation.

Reply:
I attempted in the past to make the `int` an `int?` but C# doesn't support nullables as attribute arguments. Not sure if there is any workaround to this

Reply:
I think the best solution for the feedback I've received so far is to add back a parameter `Ordered` but this time only as a `bool` which serves as an indication of whether the setting should be sorted (rather than using a value of -1). Would this suffice?

Reply:
I see, didn't realise there was such a limitation. Maybe a second constructor?

Reply:
In order for `orderPosition` to be a required parameter (in a separate constructor), `description` has to be required as well. Should I make it required or set a default value for `orderPosition`?

Or, I could move `orderPosition` before `description`, but I think that would be weird